### DISCUSSION TITLE
Result.Success / Result.Fail from an async funtion call result?

### DISCUSSION THREAD

**Discussion by the-dext** [vkhorikov/CSharpFunctionalExtensions#394] [Q&A]
Hi,
Is it possible to call and await an async method, and use that result to generate a Result.Success or Result.Failure.

As opposed to using a Maybe<T>.ToResult()

Currently I'm using code like this, but I'm hoping for something cleaner

.Bind(x =>
{
    return myAsyncFunc(arg1, rg2).Or(new int[0])
          .ToResult("this can never be returned - i think");
})

**Comment by vkhorikov** [comment]
What's the desired syntax (a rough sample is fine)?
